src/mem_usage.txt

```diff
@@ -1,27 +1,27 @@
-00000678 l     F .text	00000050 _ZN10RF24Logger5errorIPK19__FlashStringHelperEEvS3_T_z.constprop.8
-000006c8 l     F .text	00000050 _ZN10RF24Logger4warnIPK19__FlashStringHelperEEvS3_T_z.constprop.7
-00000718 l     F .text	00000050 _ZN10RF24Logger5debugIPK19__FlashStringHelperEEvS3_T_z.constprop.6
-00000768 l     F .text	00000050 _ZN10RF24Logger3logIPKcEEvhPK19__FlashStringHelperT_z.constprop.5
-000007b8 l     F .text	00000050 _ZN10RF24Logger3logIPK19__FlashStringHelperEEvhS3_T_z.constprop.4
-000008fe l     F .text	0000008e _ZN20RF24StreamLogHandler14appendLogLevelEh
+00000252 l     F .text	00000006 _ZN22RF24AbstractLogHandler11setLogLevelEh
+000004c8 l     F .text	00000022 _ZN22RF24AbstractLogHandler3logEhPK19__FlashStringHelperS2_PPv
+000004ea l     F .text	00000022 _ZN22RF24AbstractLogHandler3logEhPKcS1_PPv
+0000050c l     F .text	00000050 _ZN10RF24Logger5errorEPKcS1_z.constprop.15
+008003f7 l     O .bss	00000002 rf24Logger
+0000055c l     F .text	00000050 _ZN10RF24Logger4warnEPKcS1_z.constprop.14
+000005ac l     F .text	00000050 _ZN10RF24Logger4infoEPKcS1_z.constprop.13
+000005fc l     F .text	00000050 _ZN10RF24Logger5debugEPKcS1_z.constprop.12
+0000064c l     F .text	00000050 _ZN10RF24Logger3logEhPKcS1_z.constprop.11
+0000069c l     F .text	00000050 _ZN10RF24Logger5errorEPK19__FlashStringHelperS2_z.constprop.10
+000006ec l     F .text	00000050 _ZN10RF24Logger4warnEPK19__FlashStringHelperS2_z.constprop.9
+0000073c l     F .text	00000050 _ZN10RF24Logger4infoEPK19__FlashStringHelperS2_z.constprop.8
+0000078c l     F .text	00000050 _ZN10RF24Logger5debugEPK19__FlashStringHelperS2_z.constprop.7
+000007dc l     F .text	00000050 _ZN10RF24Logger3logEhPK19__FlashStringHelperS2_z.constprop.6
+00000922 l     F .text	00000268 _ZN20RF24StreamLogHandler12appendFormatEcPPv
+00000b8a l     F .text	0000008e _ZN20RF24StreamLogHandler14appendLogLevelEh
 00800118 l     O .data	00000008 _ZL13rf24LogLevels
-0000098c l     F .text	00000268 _ZN20RF24StreamLogHandler12appendFormatEcPPv
-00000bf4 l     F .text	00000082 _ZN20RF24StreamLogHandler15appendTimestampEv
-00000c76 l     F .text	000000b2 _ZN20RF24StreamLogHandler5writeEhPK19__FlashStringHelperS2_PPv
-00000d28 l     F .text	000000a6 _ZN20RF24StreamLogHandler5writeEhPK19__FlashStringHelperPKcPPv
-00000f12 l     F .text	00000080 _GLOBAL__I_65535_0_RF24AbstractLogHandler.cpp.o.1912
-00800406 l     O .bss	00000005 rf24SerialLogHandler
-00800132 l     O .data	0000000e _ZTV20RF24StreamLogHandler
+00000c18 l     F .text	00000082 _ZN20RF24StreamLogHandler15appendTimestampEv
+00000c9a l     F .text	000000b2 _ZN20RF24StreamLogHandler5writeEhPK19__FlashStringHelperS2_PPv
+00000d4c l     F .text	000000a6 _ZN20RF24StreamLogHandler5writeEhPKcS1_PPv
+00000f36 l     F .text	00000080 _GLOBAL__I_65535_0_RF24AbstractLogHandler.cpp.o.1928
+008003f2 l     O .bss	00000005 rf24SerialLogHandler
+00800141 l     O .data	0000000e _ZTV20RF24StreamLogHandler
```

Comment:
It is not seen here but the memory usage by the library components is still the same. I just replaced the templates into a methods with well known arguments.